.github/workflows/build-and-deploy.yaml

```diff
@@ -48,24 +51,20 @@ jobs:
 
       - name: Deploy to EC2 and Start the service
         env:
-          GITHUB_REPO: ${{ github.repository }}
           PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
           HOSTNAME: ${{secrets.SSH_HOST}}
-          branch: ${{ steps.extract_branch.outputs.branch }}
         run: |
           echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
-          ssh -o StrictHostKeyChecking=no -i private_key ubuntu@${HOSTNAME} GITHUB_REPO=${GITHUB_REPO} branch=${branch} '
-            cd /home/ubuntu/ &&
-            rm -rf basin &&
-            echo "Pulling from git@github.com:${GITHUB_REPO}.git" &&
-            git clone git@github.com:${GITHUB_REPO}.git &&
-            cd basin &&
-            git checkout ${branch} &&
+          scp -o StrictHostKeyChecking=no -i private_key docker-compose.yaml ubuntu@${HOSTNAME}:/home/ubuntu/
+          ssh -o StrictHostKeyChecking=no -i private_key ubuntu@${HOSTNAME} '
```

Comment:
Instead of git clone, only pushed the docker-compose to the EC2, and spin up the service using pre-built image